source/opt/instruction.h

```diff
@@ -241,6 +242,18 @@ class Instruction : public utils::IntrusiveNodeBase<Instruction> {
   // Returns true if the instruction annotates an id with a decoration.
   inline bool IsDecoration();
 
+  // Returns true if the instruction is known to be a load from read-only
+  // memory.
+  bool IsReadOnlyLoad(IRContext* ctx) const;
+
+  // Returns the instruction that gives the base address of an address
+  // calculation.
```

Comment:
You should document the IsLoad() pre-condition

Reply:
done.